### DISCUSSION TITLE
Question/Discussion on PointerDevice and other device capabilities classes

### DISCUSSION THREAD

**Discussion by SetTrend** [microsoft/WindowsAppSDK#378] [Q&A]
At [Retrieve pointer properties](https://docs.microsoft.com/en-us/windows/uwp/design/input/identify-input-devices#retrieve-pointer-properties) it reads:

> Note  The values returned by the properties discussed here are based on all detected pointer devices: Boolean properties return non-zero if at least one device supports a specific capability, and numeric properties return the maximum value exposed by any one pointer device.
<br/>

So ...

```c#
IReadOnlyList<PointerDevice> pointerDevices = Windows.Devices.Input.PointerDevice.GetPointerDevices();
```
... returns a list of [`PointerDevice`](https://docs.microsoft.com/en-us/uwp/api/Windows.Devices.Input.PointerDevice) objects, and each and every object in this list returns the same information?

<br/>
<br/>

- Why should it make sense to provide a list if each item in the list always returns the same values?

- Why do the different capabilities classes only provide a union set of all a device class' features?

- Doesn't UWP provide features to identify particular pointer, keyboard, touch, mouse devices and their features?

- There are [specialized keyboards on the market](https://www.amazon.com/Portable-Financial-Accounting-Keyboard-Extensions/dp/B07JVVNXQP), servicing special purposes. Shouldn't the [`KeyboardCapabilities`](https://docs.microsoft.com/en-us/uwp/api/Windows.Devices.Input.KeyboardCapabilities) class provide more properties than just [`KeyboardPresent`](https://docs.microsoft.com/en-us/uwp/api/windows.devices.input.keyboardcapabilities.keyboardpresent). For accounting, for example, it makes a difference whether a numeric keypad is available or not.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 1641e1a8-9b76-2985-3e12-49e20b79581d
* Version Independent ID: 30097945-bda2-fd54-9f39-871a79aac7f8
* Content: [Identify input devices - UWP applications](https://docs.microsoft.com/en-us/windows/uwp/design/input/identify-input-devices#retrieve-pointer-properties)
* Content Source: [windows-apps-src/design/input/identify-input-devices.md](https://github.com/MicrosoftDocs/windows-uwp/blob/docs/windows-apps-src/design/input/identify-input-devices.md)
* Product: **uwp**
* Technology: **design**
* GitHub Login: @Karl-Bridge-Microsoft
* Microsoft Alias: **kbridge**

**Comment by StephenLPeters** [comment]
PointerDevice is a part of Windows.UI.dll so I don't think that WinUI owns this? I'm moving it to project reunion but if that is wrong feel free to put it back on us and I'll find the right owner.
